src/encoded/visualization.py

```diff
@@ -2241,39 +2243,57 @@ def browsers_available(assemblies, status, files, types, item_type=None):
     elif item_type not in VISIBLE_DATASET_TYPES_LC:
             return []
 
-    if not files:
-        return []
-
     browsers = set()
     for assembly in assemblies:
-        mapped_assembly = _ASSEMBLY_MAPPER_FULL[assembly]
+        mapped_assembly = _ASSEMBLY_MAPPER_FULL.get(assembly)
         if not mapped_assembly:
             continue
-        if 'ucsc_assembly' in mapped_assembly:
-            browsers.add('ucsc')
-        if 'ensembl_host' in mapped_assembly:
-            browsers.add('ensembl')
-        if 'quickview' in mapped_assembly:
-            browsers.add('quickview')
-
-    if status not in VISIBLE_DATASET_STATUSES:
-        if status not in QUICKVIEW_STATUSES_BLOCKED:
-            return ["quickview"]
-        return []
+        vis_blob = None
+        if request is not None and accession is not None and status == 'released':
+            vis_blob = get_from_es(request, accession + "_" + assembly)  # use of find_or_make_acc_composite() will recurse!
+        if 'ucsc' not in browsers and 'ucsc_assembly' in mapped_assembly.keys():
```

Comment:
If we're going to check if the value is in the browser set then we should just make browser a list.  The benefit of using a set it that sets do not allow duplicate values.  So this check is done again when browser.add() is called.